rs/consensus/src/consensus/validator.rs

```diff
@@ -990,6 +990,20 @@ impl Validator {
                 continue;
             }
 
+            // If there exists an equivocation proof for the signer of the
+            // proposal at the same height, we ignore the block. The only
+            // way in which we would validate such a proposal is through
+            // the notarization fast-path.
+            if pool_reader
+                .pool()
+                .validated()
+                .equivocation_proof()
+                .get_by_height(proposal.height())
+                .any(|proof| proof.signer == proposal.signature.signer)
+            {
```

Comment:
Would it make sense to log a warning if this happens?

Reply:
Good suggestion, done.